src/renderers/dom/client/wrappers/ReactDOMOption.js

```diff
@@ -41,31 +41,38 @@ var ReactDOMOption = {
     // or missing (e.g., for <datalist>), we don't change props.selected
     var selected = null;
     if (selectValue != null) {
+      var value;
+      if ('value' in props) {
```

Comment:
`props.value != null` please – we treat null and undefined the same as omitted props for all DOM components.